File: CMBooking-backend/src/main/java/com/booking/CMBookingbackend/service/PropertyService.java
package com.booking.CMBookingbackend.service;

import com.booking.CMBookingbackend.dto.PropertyDto;
import com.booking.CMBookingbackend.entity.Property;
import com.booking.CMBookingbackend.response.VerifyResponse;
import org.springframework.data.domain.Page;
import org.springframework.stereotype.Service;

@Service
public interface PropertyService {

    PropertyDto createProperty(PropertyDto propertyDto);

    PropertyDto assignOwnerAndImageToProperty(String email, Long propertyId);

    VerifyResponse findByLocation(String location);

    Page<Property> getAllProperties(int page, int size);

    void deleteProperty(Long propertyId);
}

File: CMBooking-frontend/src/service/PropertyService.js
import axios from "axios";

const REST_API_BASE_URL = 'http://localhost:8080/api/property'; 

export const createProperty = (property) => axios.post(REST_API_BASE_URL, property);

export const assignOwnerAndImageToProperty = (userEmail, propertyId) => axios.post(`${REST_API_BASE_URL}/${userEmail}/${propertyId}`);

export const findByLocation = (location) => axios.post(REST_API_BASE_URL + '/' + location);

export async function getProperties(page = 0, size = 10) {
    return await axios.get(`${REST_API_BASE_URL}?page=${page}&size=${size}`);
} 

export const deleteProperty = (propertyId) => axios.delete(REST_API_BASE_URL + '/' + propertyId);